tests/integration-tests/network_template_builder.py

```diff
@@ -0,0 +1,254 @@
+        self.__availability_zone = self.__get_availability_zone(availability_zone)
+        self.__vpc_config = vpc_configuration
+        self.__vpc = self.__get_vpc(existing_vpc)
+        self.__vpc_subnets = vpc_configuration.subnets
+        self.__gateway_id = self.__get_gateway_id()
+        self.__create_ig = self.__template.add_condition("CreateInternetGateway", Equals(self.__gateway_id, ""))
+        self.__existing_ig = self.__template.add_condition(  # can't negate above condition with Not()
+            "ExistingInternetGateway", Not(Equals(self.__gateway_id, ""))
+        )
+
+    def __get_vpc(self, existing_vpc):
+        if existing_vpc:
+            return self.__add_parameter(name="VpcId", description="The vpc id", expected_input_type="String")
+        else:
+            return self.__build_vpc()
+
+    def __get_availability_zone(self, availability_zone):
+        if availability_zone:
+            return availability_zone
+        else:
+            return Ref(
+                self.__template.add_parameter(
+                    Parameter(
+                        "AvailabilityZone",
+                        Description="(Optional) The zone in which you want to create your subnet(s)",
+                        Type="String",
+                        Default="",
+                    )
+                )
+            )
+
+    def build(self):
+        """Build the template."""
+        self.__build_template()
+        return self.__template
+
+    def __build_template(self):
+        internet_gateway = self.__build_internet_gateway(self.__vpc)
+        nat_gateway = None
+        subnet_refs = []
+        for subnet in self.__vpc_subnets:
+            subnet_ref = self.__build_subnet(subnet, self.__vpc)
+            subnet_refs.append(subnet_ref)
+            if subnet.has_nat_gateway:
+                nat_gateway = self.__build_nat_gateway(subnet, subnet_ref)
+
+        for subnet, subnet_ref in zip(self.__vpc_subnets, subnet_refs):
+            self.__build_route_table(subnet, subnet_ref, self.__vpc, internet_gateway, nat_gateway)
+
+    def __build_vpc(self):
+        vpc = self.__template.add_resource(
+            VPC(
+                self.__vpc_config.name,
+                CidrBlock=self.__vpc_config.cidr,
+                EnableDnsSupport=self.__vpc_config.enable_dns_support,
+                EnableDnsHostnames=self.__vpc_config.enable_dns_hostnames,
+                Tags=self.__vpc_config.tags,
+            )
+        )
+        self.__template.add_output(Output("VpcId", Value=Ref(vpc), Description="The Vpc Id"))
+        return vpc
+
+    def __build_internet_gateway(self, vpc: VPC):
+        internet_gateway = self.__template.add_resource(
+            InternetGateway(
+                "InternetGateway",
+                Tags=Tags(Name=TAGS_PREFIX + "IG", Stack=Ref("AWS::StackId")),
+                Condition=self.__create_ig,
+            )
+        )
+        self.__template.add_resource(
```

Comment:
not sure if the rebase already took into account https://github.com/aws/aws-parallelcluster/pull/1219 but we should make sure we apply these changes